mm2src/coins/eth.rs

```diff
@@ -3011,7 +3036,7 @@ impl EthCoin {
     fn send_hash_time_locked_payment(&self, args: SendPaymentArgs<'_>) -> EthTxFut {
         let receiver_addr = try_tx_fus!(addr_from_raw_pubkey(args.other_pubkey));
         let swap_contract_address = try_tx_fus!(args.swap_contract_address.try_to_address());
-        let id = self.etomic_swap_id(args.time_lock, args.secret_hash);
+        let id = self.etomic_swap_id(args.time_lock, args.secret_hash, self.my_address);
```

Comment:
This breaks backward compatibility, I know that we discussed this in DM and we agreed that it will require a lot of code in negotiation messages to maintain backward compatibility that will not be needed later (making our code a lot worse and harder to fix later) and we can accept that old nodes can't swap with new nodes but after thinking more about this, this will lead to also funds locked for sometimes for new nodes since old nodes will not be able to validate this payment. We can add a new payment instruction probably to avoid using the ever growing negotiation messages but with the watchers payment instructions it gets more complicated too. We need to add this code and comment that we can remove it later in the future but I need to hear your opinion on what is the best way to handle this.

P.S. GUIs don't use the latest version at the same time, for instance desktop and mobile can sometimes lag behind webDEX/KomodoWallet and vise versa.